design/option_override.md

```diff
@@ -0,0 +1,53 @@
+These tables describe the desired behavior for different override situations (meant to be a proposal), and lists the required changes to the existing implementation (incomplete).
+
+Next steps:
+* Agree on desired behavior
+* Determine changes to existing implementation
+
+**Options with connection context:**
+
+Kinds of overrides:
+* "conn -> opt" - If the --name option is specified on a command, what should happen when the option in question is specified on the same command. This applies to both command line commands and interactive session commands.
+* "cmd -> int" - If there is a connection inherited from the command line to an interactive session, what should happen when the option in question is specified on an interactive command in that session. This applies regardless of how the connection at the command line level was crafted.
+
+The 'desired' columns describe the desired behavior:
+* 'update PS' means to update the currently used PywbemServer object from the option.
+* 'reset+update PS' means to reset the currently used PywbemServer object to default values and then to update it from the option. This is processed as the first option.
+* 'replace PS' means to update the currently used PywbemServer object from the new connection, replacing all connection attributes.
+
+The 'changes' columns describe the necessary changes to the existing implementation, in order to have the desired behavior.
+
+| type | general option name | desired conn -> opt | desired cmd -> int | changes conn -> opt | changes cmd -> int |
```

Comment:
Ahha. type is the connection type (real or mock)

Reply:
will do